### ISSUE TITLE
Apps break if there is no index

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19515]
If the ES index is down, we are unable to get to the apps.  The portlet frame loads, but nothing after that.

This is probably happening because the host list we are pulling back is coming from a ES query.  We need to get the host list from the db.  There is a method to do this:

`HostAPI.findAllFromDB`

As a general rule, we need to stop relying on the Index for our back end operations.  The Index needs to be for content search-ability and delivery only.

So, a test, if there is no index (deactivate your indexes), the saml app should 
1. work and allow for remote logins and 
2. should still be able to be configured.

**Comment by wezell** [comment]
```
16:47:18.372  INFO  caffine.CaffineCache - ***	 Building Cache : parentpermissionablecache, size:10000,Concurrency:32
16:47:24.025  ERROR apps.AppsResource - Exception on listing all available apps.
java.lang.NullPointerException: index must not be null
	at java.util.Objects.requireNonNull(Objects.java:228) ~[?:1.8.0_162]
	at org.elasticsearch.action.search.SearchRequest.validateIndices(SearchRequest.java:332) ~[elasticsearch-7.3.2.jar:7.3.2]
	at org.elasticsearch.action.search.SearchRequest.indices(SearchRequest.java:324) ~[elasticsearch-7.3.2.jar:7.3.2]
	at com.dotcms.content.elasticsearch.business.ESContentFactoryImpl.indexSearch(ESContentFactoryImpl.java:1660) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.searchIndex(ESContentletAPIImpl.java:973) ~[classes/:?]
	at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.searchIndex(ContentletAPIInterceptor.java:1359) ~[classes/:?]
	at com.dotcms.security.apps.AppsAPIImpl.getValidSites(AppsAPIImpl.java:143) ~[classes/:?]
	at com.dotcms.security.apps.AppsAPIImpl.appKeysByHost(AppsAPIImpl.java:163) ~[classes/:?]
	at com.dotcms.security.apps.AppsAPIImpl.appKeysByHost(AppsAPIImpl.java:132) ~[classes/:?]
	at com.dotcms.rest.api.v1.apps.AppsHelper.getAvailableDescriptorViews(AppsHelper.java:102) ~[classes/:?]
	at com.dotcms.rest.api.v1.apps.AppsResource.listAvailableApps(AppsResource.java:86) ~[classes/:?]
```

**Comment by nollymar** [comment]
Needs work: I got a NPE when I configured secrets for Google Translate and SAML.

**Steps to reproduce:** 
1. Create secrets for Google Translate and SAML
2. Delete or deactivate live and working indexes
3. Go to the Apps portlet and try to open the configured secrets.

This is the log: https://gist.github.com/nollymar/019d6fa7824a4e648b57d75a4fa4fb4f

Note: This error doesn't happen when the ES is down
